src/Middleware/Localization/src/RequestLocalizationMiddleware.cs

```diff
@@ -126,7 +126,10 @@ public async Task Invoke(HttpContext context)
 
             if (_options.ApplyCurrentCultureToResponseHeaders)
             {
-                context.Response.Headers.Add(HeaderNames.ContentLanguage, requestCulture.UICulture.Name);
+                var headers = context.Response.Headers;
+                // Can't use += as StringValues does not override operator+
+                // and the implict conversions will cause an incorrect string concat https://github.com/dotnet/runtime/issues/52507
+                headers.ContentLanguage = StringValues.Concat(headers.ContentLanguage, requestCulture.UICulture.Name);
```

Comment:
This is a behavior change, `Add` throws for duplicates, I assume the change is fine but want to call it out.
+@Tratcher for your opinion

Reply:
Most uses of Add are a mistake anyways, headers should always be Set or Append'd. Let's change this to Set to keep it consistent.